client/proxy.go

```diff
@@ -0,0 +1,1055 @@
+		if line.startswith("#") or not '$' in line:
+			out_list.append(line)
+		else:
+			out_list.append(os.path.expandvars(line))
+
+with open(filename, 'w') as f:
+	for out in out_list:
+		f.write(out)
+`
+	return expand
+}
+
+func launchScript(info UnitInfo) string {
+	launch := `
+#!/bin/sh
+
+if result=$(command -v qdrouterd 2>&1); then
+    qdr_bin=$result
+else
+    echo "qdrouterd could not be found. Please 'dnf install qpid-dispatch-router'"
+    exit
+fi
+
+proxy_name={{.ProxyName}}
+
+share_dir=${XDG_DATA_HOME:-~/.local/share}
+config_dir=${XDG_CONFIG_HOME:-~/.config}
+
+certs_dir=$share_dir/skupper/$proxy_name/qpid-dispatch-certs
+qdrcfg_dir=$share_dir/skupper/$proxy_name/config
+
+export ROUTER_ID=$(uuidgen)
+export QDR_CONF_DIR=$share_dir/skupper/$proxy_name
+export QDR_BIN_PATH=${QDROUTERD_HOME:-$qdr_bin}
+
+mkdir -p $config_dir/systemd/user
+mkdir -p $qdrcfg_dir
+mkdir -p $certs_dir
+
+cp -R ./qpid-dispatch-certs/* $certs_dir
+cp ./service/$proxy_name.service $config_dir/systemd/user/
+cp ./config/qdrouterd.json $qdrcfg_dir
+
+python3 ./expandvars.py $config_dir/systemd/user/$proxy_name.service
+python3 ./expandvars.py $qdrcfg_dir/qdrouterd.json
+
+systemctl --user enable $proxy_name.service
+systemctl --user daemon-reload
+systemctl --user start $proxy_name.service
+
+`
+	var buf bytes.Buffer
+	launchScript := template.Must(template.New("launchScript").Parse(launch))
+	launchScript.Execute(&buf, info)
+
+	return buf.String()
+}
+
+func removeScript(info UnitInfo) string {
+	remove := `
+#!/bin/sh
+
+proxy_name={{.ProxyName}}
+
+share_dir=${XDG_DATA_HOME:-~/.local/share}
+config_dir=${XDG_CONFIG_HOME:-~/.config}
+
+systemctl --user stop $proxy_name.service
+systemctl --user disable $proxy_name.service
+systemctl --user daemon-reload
+
+rm -rf $share_dir/skupper/$proxy_name
+rm $config_dir/systemd/user/$proxy_name.service
+`
+	var buf bytes.Buffer
+	removeScript := template.Must(template.New("removeScript").Parse(remove))
+	removeScript.Execute(&buf, info)
+
+	return buf.String()
+}
+
+func getDataHome() string {
+	dataHome, ok := os.LookupEnv("XDG_DATA_HOME")
+	if !ok {
+		homeDir, _ := os.UserHomeDir()
+		return homeDir + "/.local/share"
+	} else {
+		return dataHome
+	}
+}
+
+func getConfigHome() string {
+	configHome, ok := os.LookupEnv("XDG_CONFIG_HOME")
+	if !ok {
+		homeDir, _ := os.UserHomeDir()
+		return homeDir + "/.config"
+	} else {
+		return configHome
+	}
+}
+
+func newUUID() string {
+	return uuid.New().String()
+}
+
+func isActive(proxyName string) bool {
+	cmd := exec.Command("systemctl", "--user", "check", proxyName)
+	err := cmd.Run()
+	if err == nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+func generateProxyName(namespace string, cli kubernetes.Interface) (string, error) {
+	proxies, err := cli.CoreV1().ConfigMaps(namespace).List(metav1.ListOptions{LabelSelector: "skupper.io/type=proxy-definition"})
+	max := 1
+	if err == nil {
+		proxy_name_pattern := regexp.MustCompile("proxy([0-9])+")
```

Comment:
This has the same issue as Fernando pointed out for the link/connector name generation (fixed in my claims PR). The regex should be proxy([0-9]+)+ in order to match multi digit values (unusual for proxy, but worth fixing). Also the label selector will only look at proxy configmaps, so a configmap that is not that type but clashes with the name would block new proxies.